c10/util/StringUtil.cpp

```diff
@@ -33,18 +33,33 @@ std::ostream& operator<<(std::ostream& out, const SourceLocation& loc) {
   return out;
 }
 
-size_t ReplaceAll(std::string& s, const char* from, const char* to) {
-  TORCH_CHECK(from && *from, "");
-  TORCH_CHECK(to, "");
+size_t ReplaceAll(std::string& s, c10::string_view from, c10::string_view to) {
+  if (from.empty()) {
+    return 0;
+  }
 
   size_t numReplaced = 0;
-  std::string::size_type lenFrom = std::strlen(from);
-  std::string::size_type lenTo = std::strlen(to);
-  for (auto pos = s.find(from); pos != std::string::npos;
-       pos = s.find(from, pos + lenTo)) {
-    s.replace(pos, lenFrom, to);
-    numReplaced++;
+  std::string::size_type last_pos = 0u;
+  std::string::size_type cur_pos = 0u;
+  const c10::string_view input(s);
+  std::string buffer;
```

Comment:
I almost requested changes on this because I thought in-place replacement was more efficient, but it's quadratic time as implemented since we have to shift O(N) characters on each replace() call. Nice fix!

Perhaps you could follow up to use the existing string as the buffer if the replacement string is not longer than the string to be replaced?